authorization/permission/service/permission_service.go

```diff
@@ -31,7 +32,7 @@ func NewPermissionModelService(db *gorm.DB) PermissionModelService {
 // parent and other ancestor resources, and also takes into account role mappings, which allow roles assigned for a
 // certain type of resource in the resource ancestry to map to a role for a different resource type lower in the
 // resource hierarchy.
-func (s *GormPermissionModelService) HasScope(ctx context.Context, identityID uuid.UUID, resourceID string, scopeName string) (bool, error) {
+func (s *PermissionServiceImpl) HasScope(ctx context.Context, identityID uuid.UUID, resourceID string, scopeName string) (bool, error) {
```

Comment:
Same thing. You are mixing Model Service (need a better name) with Controller Service.
You made this code a controller service. But then it must use transactions.
Just move this code back to the Model Service where it was before :)